pkg/operator/reports/name.go

```diff
@@ -0,0 +1,36 @@
+package reports
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aquasecurity/trivy-operator/pkg/kube"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func NameFromController(controller client.Object) string {
+	kind := controller.GetObjectKind().GroupVersionKind().Kind
+	name := controller.GetName()
+	reportName := fmt.Sprintf("%s-%s", strings.ToLower(kind), name)
+	if isValidName(reportName) {
+		return reportName
+	}
+	return fmt.Sprintf("%s-%s", strings.ToLower(kind), kube.ComputeHash(name))
+}
+
+func NameFromControllerContainer(controller client.Object, container string) string {
+	kind := controller.GetObjectKind().GroupVersionKind().Kind
+	name := controller.GetName()
+	reportName := fmt.Sprintf("%s-%s-%s", strings.ToLower(kind), name, container)
+	if isValidName(reportName) {
+		return reportName
+	}
+
+	return fmt.Sprintf("%s-%s", strings.ToLower(kind), kube.ComputeHash(name+"-"+container))
```

Comment:
```suggestion
        return fmt.Sprintf("%s-%s", strings.ToLower(kind), kube.ComputeHash(fmt.Sprintf("%s-%s", name, container)))

```

Reply:
I don't think we should use fmt here, and leave it as it is. It would be better if the `ComputeHash` func accepted a varadic argument, but that would be a breaking change..

Reply:
explain why not?

Reply:
> explain why not use fmt?

1. Because the code was not using `fmt` before this refactoring (and fix)
2. Using `fmt.Sprintf` involves a lot of code for performing advanced formatting that is not needed here. The purpose here is to get something unique from the `name` and `container` values.

Reply:
IMO it make sense to use formatting , it make it more clean, clear (even if it wan't like it before) , same as we use it other places in the code rather than connecting with `+`.

Reply:
wdyt about hashing the `reportName` instead of constructing a new one? eg: `kube.ComputeHash(reportName)`
we would be hashing the "real" name that was invalid, but I think the hash wouldn't break because of it.